.github/workflows/main.yaml

```
@@ -6,22 +6,36 @@ on:
       - main
   push:
   schedule:
-    - cron: '20 7 * * 0'
+    - cron: '40 6 * * 0'
 
 jobs:
+  check-libraries:
+    uses: canonical/sdcore-github-workflows/.github/workflows/check-libraries.yaml@v1.0.0
+    secrets:
+      CHARMCRAFT_AUTH: ${{ secrets.CHARMCRAFT_AUTH }}
+
+
   lint-report:
-    uses: canonical/sdcore-github-workflows/.github/workflows/lint-report.yaml@v0.0.1
+    uses: canonical/sdcore-github-workflows/.github/workflows/lint-report.yaml@v1.0.0
 
   static-analysis:
-    uses: canonical/sdcore-github-workflows/.github/workflows/static-analysis.yaml@v0.0.1
+    uses: canonical/sdcore-github-workflows/.github/workflows/static-analysis.yaml@v1.0.0
 
   unit-tests-with-coverage:
-    uses: canonical/sdcore-github-workflows/.github/workflows/unit-test.yaml@v0.0.1
+    uses: canonical/sdcore-github-workflows/.github/workflows/unit-test.yaml@v1.0.0
+
+  build:
+    needs:
+      - lint-report
+      - static-analysis
+      - unit-tests-with-coverage
+    uses: canonical/sdcore-github-workflows/.github/workflows/build.yaml@v1.0.0
+    secrets: inherit
 
   integration-test:
-    uses: ./.github/workflows/integration-tests.yaml
-    with:
-      charm-file-name: "sdcore-upf_ubuntu-22.04-amd64.charm"
+    needs:
+      - build
+    uses: canonical/sdcore-github-workflows/.github/workflows/integration-test.yaml@v1.0.0
```

This one took me way to long to nail down. It needs to be using its own local copy of integration-test.yaml because that one is for K8s, not machine, charms.
```suggestion
    uses: ./.github/workflows/integration-tests.yaml
```

---

Nice catch!